bravado/response.py

```diff
@@ -1,24 +1,39 @@
 # -*- coding: utf-8 -*-
 import monotonic
+import typing
+from bravado_core.response import IncomingResponse  # noqa: F401
 
+if typing.TYPE_CHECKING:  # Needed to avoid cyclic import.
+    from bravado.config import RequestConfig  # noqa: F401
 
-class BravadoResponse(object):
+
+R = typing.TypeVar('R')
+T = typing.TypeVar('T')
+
+
+class BravadoResponse(typing.Generic[T]):
     """Bravado response object containing the swagger result as well as response metadata.
 
     :ivar result: Swagger result from the server
     :ivar BravadoResponseMetadata metadata: metadata for this response including HTTP response
     """
 
-    def __init__(self, result, metadata):
+    def __init__(
+        self,
+        result,  # type: R
```

Comment:
Shouldn't this be T? Also, if you need R then probably the class should be `Generic[R, T]`.